### PR TITLE
SubscriptionUniqueRootField validation to check multiple fragments - Validation Bug fix 

### PR DESCRIPTION
**Pull request by Juliano-Prado**
Changing SubscriptionUniqueRootField validation in order to also check inline fragments and multiple fragments
Fix for: https://github.com/graphql-java/graphql-java/issues/3425

### CODE REVIEW

File: src/test/groovy/graphql/validation/rules/SubscriptionUniqueRootFieldTest.groovy [hunk 1]
```diff
@@ -151,9 +151,141 @@ class SubscriptionUniqueRootFieldTest extends Specification {
         !validationErrors.empty
         validationErrors.size() == 1
         validationErrors[0].validationErrorType == ValidationErrorType.SubscriptionIntrospectionRootField
-        validationErrors[0].message == "Validation error (SubscriptionIntrospectionRootField) : Subscription operation 'doggo' fragment root field '__typename' cannot be an introspection field"
+        validationErrors[0].message == "Validation error (SubscriptionIntrospectionRootField) : Subscription operation 'doggo' root field '__typename' cannot be an introspection field"
+    }
+
+    def "5.2.3.1 subscription with multiple root fields within inline fragment are not allowed"() {
```
Comment (Juliano-Prado) [inline src/test/groovy/graphql/validation/rules/SubscriptionUniqueRootFieldTest.groovy:157]: add this test that was failing before this PR changes.

File: src/test/groovy/graphql/validation/rules/SubscriptionUniqueRootFieldTest.groovy [hunk 2]
```diff
@@ -151,9 +151,141 @@ class SubscriptionUniqueRootFieldTest extends Specification {
...
+        then:
+        !validationErrors.empty
+        validationErrors.size() == 1
+        validationErrors[0].validationErrorType == ValidationErrorType.SubscriptionMultipleRootFields
+        validationErrors[0].message == "Validation error (SubscriptionMultipleRootFields) : Subscription operation 'whoIsAGoodBoy' must have exactly one root field"
     }
 
+
+    def "5.2.3.1 subscription with more than one root field with multiple fragment with inline fragments fails validation"() {
+        given:
```
Comment (Juliano-Prado) [inline src/test/groovy/graphql/validation/rules/SubscriptionUniqueRootFieldTest.groovy:219]: add this test that was failing before this PR changes.

File: src/test/groovy/graphql/validation/rules/SubscriptionUniqueRootFieldTest.groovy [hunk 3]
```diff
@@ -151,9 +151,141 @@ class SubscriptionUniqueRootFieldTest extends Specification {
...
+        then:
+        !validationErrors.empty
+        validationErrors.size() == 1
+        validationErrors[0].validationErrorType == ValidationErrorType.SubscriptionMultipleRootFields
+        validationErrors[0].message == "Validation error (SubscriptionMultipleRootFields) : Subscription operation 'whoIsAGoodBoy' must have exactly one root field"
+    }
+
+
+    def "5.2.3.1 subscription with one root field with multiple fragment with inline fragments does not fail validation"() {
+        given:
```
Comment (Juliano-Prado) [inline src/test/groovy/graphql/validation/rules/SubscriptionUniqueRootFieldTest.groovy:258]: Added just to ensure the valid case of multiple fragments is working fine.

**Review by bbakerman — APPROVED** [review]
Nice one

**Review by bbbiyeba — COMMENTED** [review]
Great fix, as using FieldCollector.collectFields() instead of manually traversing selections is the right approach here since it handles all fragment resolution cases uniformly without needing to special-case inline vs named fragments

Quick question: Are there any edge cases around conditional fragments with @skip or @include directives that collectFields might handle differently than expected?

**Comment by bbakerman** [comment]
> Quick question: Are there any edge cases around conditional fragments with @Skip or @include directives that collectFields might handle differently than expected?

Probably - feel free to write some unit tests and see if thats the case
